pkg/frontend/querymiddleware/split_by_instant_interval.go

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package querymiddleware
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/tenant"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/parser"
+
+	apierror "github.com/grafana/mimir/pkg/api/error"
+	"github.com/grafana/mimir/pkg/frontend/querymiddleware/astmapper"
+	"github.com/grafana/mimir/pkg/storage/lazyquery"
+)
+
+// TODO: add metrics
+
+// splitByInstantIntervalMiddleware is a Middleware that can (optionally) split the instant query by splitInterval
+type splitByInstantIntervalMiddleware struct {
```

Comment:
[nit] Would `splitInstantQueryByIntervalMiddleware` be a more clear name? @colega WDYT too?

Reply:
Done 🚀